From: PupJune2 <PupJune2@users.noreply.github.com>
Date: Sat, 17 Oct 2026 10:00:00 +0000
Subject: CromIKO Boot Animation

Show an animated CromIKO mascot on the Android 12+ splash screen: the
mascot pops in and the star twinkles in (under a second, as the
platform requires). The mascot is charcoal on the light splash and
white on the dark one; the face is cut out so it always matches the
background.

The drawable is generated by tools/cromiko/gen_drawables.py from the
logo SVG. Android 10 and 11 have no system splash screen and are not
affected.

License: GPL-3.0-only - https://spdx.org/licenses/GPL-3.0-only.html
---
 components/browser_ui/theme/android/BUILD.gn  |   1 +
 .../java/res/drawable/cromiko_splash_icon.xml | 129 ++++++++++++++++++
 .../android/java/res/values-night/colors.xml  |   3 +
 .../theme/android/java/res/values/colors.xml  |   4 +
 .../android/templates/res/values/themes.xml   |   3 +
 5 files changed, 140 insertions(+)
 create mode 100644 components/browser_ui/theme/android/java/res/drawable/cromiko_splash_icon.xml

diff --git a/components/browser_ui/theme/android/BUILD.gn b/components/browser_ui/theme/android/BUILD.gn
--- a/components/browser_ui/theme/android/BUILD.gn
+++ b/components/browser_ui/theme/android/BUILD.gn
@@ -11,6 +11,7 @@ jinja_template_resources("theme_template_resources") {
 
 android_resources("java_resources") {
   sources = [
+    "java/res/drawable/cromiko_splash_icon.xml",
     "java/res/values-night/colors.xml",
     "java/res/values-night/themes.xml",
     "java/res/values/attrs.xml",
diff --git a/components/browser_ui/theme/android/java/res/drawable/cromiko_splash_icon.xml b/components/browser_ui/theme/android/java/res/drawable/cromiko_splash_icon.xml
new file mode 100644
--- /dev/null
+++ b/components/browser_ui/theme/android/java/res/drawable/cromiko_splash_icon.xml
@@ -0,0 +1,129 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+Copyright 2026 The CromIKO Authors
+License: GPL-3.0-only - https://spdx.org/licenses/GPL-3.0-only.html
+Generated by tools/cromiko/gen_drawables.py from tools/cromiko/cromiko-logo.svg.
+-->
+
+<animated-vector
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:aapt="http://schemas.android.com/aapt">
+
+    <aapt:attr name="android:drawable">
+        <vector
+            android:width="288dp"
+            android:height="288dp"
+            android:viewportWidth="288"
+            android:viewportHeight="288">
+            <group
+                android:name="mascot"
+                android:pivotX="144"
+                android:pivotY="144">
+                <group
+                    android:translateX="-63.3526"
+                    android:translateY="-86.2011"
+                    android:scaleX="0.2797"
+                    android:scaleY="0.2797">
+                    <path
+                        android:fillType="evenOdd"
+                        android:pathData="M803.93 593.11 C838.39 589.85 851.01 624.09 864.97 649.82 C884.68 685.41 902.32 722.11 917.8 759.73 C925.28 777.9 931.31 796.56 938.78 814.54 C936.01 817.99 931.67 822.69 928.52 825.7 C839.78 910.44 685.13 973.41 563.18 972.11 C534.86 971.05 495.27 965.19 475.34 943.51 C466.31 933.8 461.67 920.82 462.5 907.59 C464.55 870.78 500.89 834.18 527.1 811.09 C525.11 774.06 520.67 696.72 539.67 664.53 C564.4 622.62 616.8 661.34 645.13 679.34 C677.99 666.89 709.82 656.58 744.25 649.57 C762.72 629.72 777.86 599.34 803.93 593.11 Z M797.41 727.23 C848.2 720.94 860.38 755.72 871.47 795.57 C878.96 822.49 871.64 848.19 847.4 863.79 C833.33 872.84 807.12 879.43 791.41 884.51 C762.85 893.98 734.15 903.01 705.3 911.59 C669.3 919.28 641.37 902.72 633.44 866.84 C627.32 839.19 626.05 816.23 641.11 790.84 C655.96 765.79 679.85 762.23 705.24 753.6 C735.79 743.44 766.03 734.47 797.41 727.23 Z M814.02 766.18 C830.19 766.36 833.9 783.97 838.69 796.38 C844.23 810.73 847.51 825.01 830.19 831.19 C818.34 830.43 813.9 824.25 810.13 813.43 C804.41 796.99 791.12 774.86 814.02 766.18 Z M721.03 796.19 C729.88 795.12 738.03 798.5 741.22 807.04 C747.4 823.57 763.84 848.59 742.5 859.49 C721.79 862.61 721.57 841.99 715.01 827.2 C709.12 813.9 707.11 804.88 721.03 796.19 Z M528.61 833.78 C529.04 834.23 538.96 885.73 541.55 893.73 C547.09 910.85 552.48 923.54 560.45 940.09 C550 939.07 542.66 937.76 532.54 935.42 C473.97 915.88 495.05 867.32 528.61 833.78 Z M740.42 672.1 C742.82 671.69 746.1 671.21 748.47 672.04 C750.81 672.88 752.7 674.67 753.68 676.96 C754.66 679.26 754.61 681.62 753.48 683.87 C748.49 693.82 699.81 703.99 687.43 707.86 C684.67 708.42 681.73 708.54 679.11 707.39 C676.57 706.27 674.68 704.21 673.86 701.56 C673.26 699.65 673.46 697.65 674.47 695.89 C680.09 686.09 727.36 675.82 740.42 672.1 Z"
+                        android:fillColor="@color/cromiko_mascot_color"/>
+                    <path
+                        android:pathData="M928.66 702.48 C929.31 702.53 929.97 702.58 930.62 702.65 C958.31 705.41 988.37 713.72 1006.68 735.9 C1046.14 790.42 989.39 843.43 953.83 879.86 C956.98 892.57 961.98 909.37 961.5 922.2 C960.32 953.38 939.16 964.4 911.52 963.76 C909.23 972.07 906.62 980.28 903.7 988.39 C892.25 1019.81 866.77 1063.68 823.8 1050.55 C809.47 1046.17 802.55 1025.85 797.09 1012.93 C736.01 1060.33 611.55 1072.24 561.65 1004.06 C572.23 1005.29 591.03 1004.35 601.99 1003.17 C695.26 993.1 796.68 951.13 873.87 898.75 C914.18 869.7 969.81 827.4 982.11 776.91 C990.37 743.06 957.07 726.34 928.82 721.65 L928.66 702.48 Z"
+                        android:fillColor="@color/cromiko_mascot_color"/>
+                    <path
+                        android:pathData="M1006.68 735.9 C1046.14 790.42 989.39 843.43 953.83 879.86 L938.4 892.46 C934.96 888.54 931.41 884.15 928.04 880.12 C966.17 847.37 1027.42 791.53 1006.68 735.9 Z"
+                        android:fillColor="@color/cromiko_accent_color"/>
+                </group>
+                <group
+                    android:name="star"
+                    android:pivotX="196.2935"
+                    android:pivotY="96.1673">
+                    <group
+                        android:translateX="-63.3526"
+                        android:translateY="-86.2011"
+                        android:scaleX="0.2797"
+                        android:scaleY="0.2797">
+                        <path
+                            android:pathData="M928 605.54 C929.99 607.52 936.6 627.46 939.22 631.81 C946.04 643.17 964.61 647.94 976.55 651.07 C943.29 660.87 938.05 665.19 928.72 698.31 C915.91 663.98 917.43 661.01 879.8 651.48 C912.83 641.16 918.04 638.91 928 605.54 Z"
+                            android:fillColor="@color/cromiko_accent_color"/>
+                    </group>
+                </group>
+            </group>
+        </vector>
+    </aapt:attr>
+
+    <target android:name="mascot">
+        <aapt:attr name="android:animation">
+            <set>
+                <objectAnimator
+                    android:propertyName="scaleX"
+                    android:valueFrom="0.7"
+                    android:valueTo="1"
+                    android:valueType="floatType"
+                    android:duration="600"
+                    android:interpolator="@android:interpolator/overshoot"/>
+                <objectAnimator
+                    android:propertyName="scaleY"
+                    android:valueFrom="0.7"
+                    android:valueTo="1"
+                    android:valueType="floatType"
+                    android:duration="600"
+                    android:interpolator="@android:interpolator/overshoot"/>
+                <objectAnimator
+                    android:propertyName="rotation"
+                    android:valueFrom="-10"
+                    android:valueTo="0"
+                    android:valueType="floatType"
+                    android:duration="600"
+                    android:interpolator="@android:interpolator/decelerate_cubic"/>
+            </set>
+        </aapt:attr>
+    </target>
+
+    <target android:name="star">
+        <aapt:attr name="android:animation">
+            <set>
+                <objectAnimator
+                    android:duration="750">
+                    <propertyValuesHolder
+                        android:propertyName="scaleX"
+                        android:valueType="floatType">
+                        <keyframe android:fraction="0" android:value="0"/>
+                        <keyframe android:fraction="0.4" android:value="0"/>
+                        <keyframe
+                            android:fraction="1"
+                            android:value="1"
+                            android:interpolator="@android:interpolator/overshoot"/>
+                    </propertyValuesHolder>
+                </objectAnimator>
+                <objectAnimator
+                    android:duration="750">
+                    <propertyValuesHolder
+                        android:propertyName="scaleY"
+                        android:valueType="floatType">
+                        <keyframe android:fraction="0" android:value="0"/>
+                        <keyframe android:fraction="0.4" android:value="0"/>
+                        <keyframe
+                            android:fraction="1"
+                            android:value="1"
+                            android:interpolator="@android:interpolator/overshoot"/>
+                    </propertyValuesHolder>
+                </objectAnimator>
+                <objectAnimator
+                    android:duration="750">
+                    <propertyValuesHolder
+                        android:propertyName="rotation"
+                        android:valueType="floatType">
+                        <keyframe android:fraction="0" android:value="-90"/>
+                        <keyframe android:fraction="0.4" android:value="-90"/>
+                        <keyframe
+                            android:fraction="1"
+                            android:value="0"
+                            android:interpolator="@android:interpolator/decelerate_cubic"/>
+                    </propertyValuesHolder>
+                </objectAnimator>
+            </set>
+        </aapt:attr>
+    </target>
+</animated-vector>
diff --git a/components/browser_ui/theme/android/java/res/values-night/colors.xml b/components/browser_ui/theme/android/java/res/values-night/colors.xml
--- a/components/browser_ui/theme/android/java/res/values-night/colors.xml
+++ b/components/browser_ui/theme/android/java/res/values-night/colors.xml
@@ -9,4 +9,7 @@ found in the LICENSE file.
     <!-- This points to a system color resource in API 31+. However, it's unclear what its value is
          on devices that don't support dynamic colors. -->
     <color name="splash_screen_bg">@color/system_neutral1_900</color>
+
+    <!-- CromIKO: the mascot is white on dark backgrounds. -->
+    <color name="cromiko_mascot_color">#FFFFFF</color>
 </resources>
diff --git a/components/browser_ui/theme/android/java/res/values/colors.xml b/components/browser_ui/theme/android/java/res/values/colors.xml
--- a/components/browser_ui/theme/android/java/res/values/colors.xml
+++ b/components/browser_ui/theme/android/java/res/values/colors.xml
@@ -9,4 +9,8 @@ found in the LICENSE file.
     <!-- This points to a system color resource in API 31+. However, it's unclear what its value is
          on devices that don't support dynamic colors. -->
     <color name="splash_screen_bg">@color/system_neutral1_10</color>
+
+    <!-- CromIKO: colors of the animated mascot (splash and welcome screens). -->
+    <color name="cromiko_mascot_color">#1E2024</color>
+    <color name="cromiko_accent_color">#5C7CFC</color>
 </resources>
diff --git a/components/browser_ui/theme/android/templates/res/values/themes.xml b/components/browser_ui/theme/android/templates/res/values/themes.xml
--- a/components/browser_ui/theme/android/templates/res/values/themes.xml
+++ b/components/browser_ui/theme/android/templates/res/values/themes.xml
@@ -209,6 +209,9 @@ Note: This is a jinja2 template, processed at build time into the final resource
         <!-- Window Properties -->
         <item name="android:windowBackground">@macro/default_bg_color</item>
         <item name="android:windowSplashScreenBackground" tools:targetApi="31">@color/splash_screen_bg</item>
+        <!-- CromIKO: animated mascot on the Android 12+ splash screen. -->
+        <item name="android:windowSplashScreenAnimatedIcon" tools:targetApi="31">@drawable/cromiko_splash_icon</item>
+        <item name="android:windowSplashScreenAnimationDuration" tools:targetApi="31">800</item>
 
         <!-- Status bar color -->
         <item name="android:statusBarColor">@macro/default_bg_color</item>
--
